packages/webapp/src/containers/LocationDetails/PointDetails/SensorDetail/v2/PostSensor.tsx

```diff
@@ -16,51 +16,59 @@ import { useTranslation } from 'react-i18next';
 import { useDispatch } from 'react-redux';
 import { History } from 'history';
 import { ContextForm, Variant } from '../../../../../components/Form/ContextForm';
+import Partners from './Partners';
 import PageTitle from '../../../../../components/PageTitle/v2';
 import { enqueueErrorSnackbar } from '../../../../Snackbar/snackbarSlice';
+import { useAddFarmAddonMutation, useLazyGetSensorsQuery } from '../../../../../store/api/apiSlice';
+import { type AddSensorsFormFields, FarmAddonField, PARTNER } from './types';
+import { ESCI_PARTNER_ID } from './constants';
 import styles from './styles.module.scss';
 
 interface PostSensorProps {
   history: History;
+  isCompactSideMenu: boolean;
 }
 
-const PostSensor = ({ history }: PostSensorProps) => {
+const PostSensor = ({ history, isCompactSideMenu }: PostSensorProps) => {
   const { t } = useTranslation();
   const dispatch = useDispatch();
 
-  const linkOrganizationId = async () => {
-    // TODO: POST /farm_addon
-    //       When failed: snackbar
+  const [addFarmAddon] = useAddFarmAddonMutation();
+  const [triggerGetSensors] = useLazyGetSensorsQuery();
 
-    // Simulating the API call
-    return new Promise<void>((resolve, reject) => {
-      setTimeout(() => {
-        // Successful
-        resolve();
+  const linkEsci = async (values: AddSensorsFormFields) => {
+    const result = await addFarmAddon(values[PARTNER]);
 
-        // Failed
-        // reject(dispatch(enqueueErrorSnackbar('TODO: Error message')));
-      }, 1500);
-    });
+    if ('error' in result) {
+      const isInvalidId = 'data' in result.error && result.error.data === 'Organisation not found';
+      const errorMessage = isInvalidId
+        ? t('SENSOR.ESCI.ORGANISATION_ID_ERROR')
+        : t('SENSOR.ESCI.ORGANISATION_ID_GENERIC_ERROR');
+      dispatch(enqueueErrorSnackbar(errorMessage));
+
+      // This error prevents the page transition in WithStepperProgressBar's onContinue
+      throw Error(isInvalidId ? 'Invalid ESCI organisation ID' : 'ESCI Connection Error');
+    }
   };
 
-  const onSave = async (data: any, onSuccess: () => void) => {
-    // TODO: GET devices with useLazyQuery.
-    //       Once the data is returned, call onSuccess to navigate to the next view.
+  const onSave = async (data: AddSensorsFormFields, onSuccess: () => void) => {
+    // Fetch sensors
+    await triggerGetSensors();
 
+    // Proceed to sensors view
     onSuccess();
   };
 
   const getFormSteps = () => [
     {
-      FormContent: () => <div>Partner selection view</div>,
-      onContinueAction: linkOrganizationId,
+      FormContent: () => <Partners />,
+      onContinueAction: linkEsci,
     },
     { FormContent: () => <div>ESCI devices view</div> },
```

Comment:
Is this temporary? I don't think ESCI devices view should be a step in the form. I see it looks that way in figma... but I am not sure its appropriate. If we are making a pattern similar to AnimalsSummary we should maybe reuse that instead and redirect to this page.